src/v/kafka/server/group_data_parser.h

```diff
@@ -101,17 +101,68 @@ class group_data_parser {
               b, group::aborted_tx_record_version);
             return handle_abort(b.header(), std::move(data));
         }
-        if (
-          b.header().type == model::record_batch_type::tx_fence
-          || b.header().type == model::record_batch_type::group_fence_tx) {
+        if (b.header().type == model::record_batch_type::tx_fence) {
+            /*
+              A potential bug in 24.2.0 resulted in a situation where tx_fence
+              batches were retained _after_ compaction while their corresponding
+              data/commit/abort batches were compacted away. This applied to
+              only group transactions that used tx_fence to begin the
+              transaction.
+
+              Historical context: tx_fence was considered historically as
+              fence batch that begins a group transaction and regular data
+              partition transaction. That changed starting 24.2.0 where a
+              dedicated fence batch type (group_tx_fence) was used for group
+              transaction fencing.
+
+              After this buggy compaction, these uncleaned tx_fence batches are
```

Comment:
Is the buggy compaction fixed now? Otherwise. shouldn't we just fix the buggy compaction instead?

Reply:
we can't fix that easily without a major refactor of the code and we don't need to (as outlined in the comment) as we already moved away from that problematic tx_fence batch infavor of group_tx_fence, so the whole fix revolves around how to get rid of the outdated tx_fence batches.